apps/server/src/core/logger/logger.service.ts

```diff
@@ -22,44 +22,35 @@ export class Logger implements ILogger {
 
 	constructor(@Inject(WINSTON_MODULE_PROVIDER) private readonly logger: WinstonLogger) {}
 
-	log(message: unknown, context?: string | undefined): void {
-		this.logger.log('info', this.createMessage(message, context));
+	log(loggable: ILoggable): void {
+		this.logger.info(LoggingUtils.createMessage(loggable, this.context));
 	}
 
-	warn(message: unknown, context?: string | undefined): void {
-		this.logger.warning(this.createMessage(message, context));
+	warn(loggable: ILoggable): void {
+		this.logger.warn(LoggingUtils.createMessage(loggable, this.context));
 	}
 
-	debug(message: unknown, context?: string | undefined): void {
-		this.logger.debug(this.createMessage(message, context));
+	debug(loggable: ILoggable): void {
+		this.logger.debug(LoggingUtils.createMessage(loggable, this.context));
 	}
 
-	verbose(message: unknown, context?: string | undefined): void {
-		this.logger.verbose(this.createMessage(message, context));
+	verbose(loggable: ILoggable): void {
+		this.logger.verbose(LoggingUtils.createMessage(loggable, this.context));
 	}
 
-	http(message: RequestLoggingBody, context?: string | undefined): void {
-		this.logger.notice(this.createMessage(message, context));
-	}
-
-	error(message: unknown, trace?: unknown, context?: string | undefined): void {
-		const result = {
-			message,
-			trace,
-		};
-		this.logger.error(this.createMessage(result, context));
+	http(loggable: ILoggable): void {
+		this.logger.notice(LoggingUtils.createMessage(loggable, this.context));
 	}
 
 	setContext(name: string) {
 		this.context = name;
 	}
+}
 
-	private createMessage(message: unknown, context?: string | undefined) {
-		return { message: this.stringifiedMessage(message), context: context || this.context };
-	}
+export class ErrorLogger implements IErrorLogger {
```

Comment:
is there an advantage of defining this as a seperate logger over one logger implementing both interfaces?

Reply:
I separated it because of the accetance criterium that we only want to log errors in the error filter. With the separation we have one logger for the filter that can only log errors and one for the rest of the code that cannot log errors.

I don't quite get, why you would split only the interface. (I'm not sure if I have some kind of misconception there...)